src/commands/unlink.js

```diff
@@ -1,16 +1,57 @@
 // @flow
+import Project from '../Project';
+import * as yarn from '../utils/yarn';
+import * as logger from '../utils/logger';
 import * as options from '../utils/options';
 import { BoltError } from '../utils/errors';
+import * as messages from '../utils/messages';
 
-export type UnlinkOptions = {};
+function getWorkspaceMap(workspaces) {
+  let workspaceMap = new Map();
+
+  for (let workspace of workspaces) {
+    workspaceMap.set(workspace.pkg.config.getName(), workspace);
+  }
+
+  return workspaceMap;
+}
+
+export type UnlinkOptions = {
+  cwd?: string,
+  packagesToUnlink: ?Array<string>
+};
 
 export function toUnlinkOptions(
   args: options.Args,
   flags: options.Flags
 ): UnlinkOptions {
-  return {};
+  return {
+    cwd: options.string(flags.cwd, 'cwd'),
+    packagesToUnlink: args
+  };
 }
 
 export async function unlink(opts: UnlinkOptions) {
-  throw new BoltError('Unimplemented command "unlink"');
+  let cwd = opts.cwd || process.cwd();
+  let packagesToUnlink = opts.packagesToUnlink;
+  let project = await Project.init(cwd);
+  let workspaces = await project.getWorkspaces();
+  let workspaceMap = getWorkspaceMap(workspaces);
+
+  // guard to check if there are packages to unlink
+  if (packagesToUnlink && packagesToUnlink.length) {
+    await Promise.all(
+      packagesToUnlink.map(async packageToUnlink => {
+        if (workspaceMap.has(packageToUnlink)) {
+          logger.warn(messages.unlinkInternalPackage(packageToUnlink));
+        } else {
+          await yarn.unlink(cwd, packageToUnlink);
+        }
+      })
+    );
+  } else {
+    throw new BoltError(
+      'Please specify package to unlink. Try: bolt w [packageToUnlink] unlink'
```

Comment:
`bolt workspace [dep] unlink`

Reply:
Actually, why scoped under `workspace/w`?

Reply:
So that we can create link only to the packages/ workspaces instead of complete project,
since other project/ repos will be dependent only on a packages. is my understanding correct?

Reply:
It's weird to me that you wouldn't want to link it in every package.

If packages A B and C all depend on X and you run `bolt workspace C link X` it seems odd that you wouldn't want to be linking it everywhere

Reply:
it will get linked everywhere in the project.